Cron/CancelOrders.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2021 Adyen NV (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Cron;
+
+use Adyen\Payment\Cron\Providers\OrdersProviderInterface;
+use Magento\Sales\Model\Order;
+use Magento\Sales\Model\Order\Status\HistoryFactory;
+use Magento\Sales\Model\Service\OrderService;
+
+class CancelOrders
+{
+    /**
+     * @var OrdersProviderInterface[]
+     */
+    private $providers;
+    /**
+     * @var OrderService
+     */
+    private $orderService;
+    /**
+     * @var HistoryFactory
+     */
+    private $orderStatusHistoryFactory;
+
+    /**
+     * ServerIpAddress constructor.
+     * @param array $providers
+     */
+    public function __construct(
+        OrderService $orderService,
+        HistoryFactory $orderStatusHistoryFactory,
+        array $providers
+    ) {
+        $this->providers = $providers;
+        $this->orderService = $orderService;
+        $this->orderStatusHistoryFactory = $orderStatusHistoryFactory;
+    }
+
+    public function execute()
+    {
+        foreach ($this->providers as $provider) {
+            foreach ($provider->provide() as $orderToCancel) {
```

Comment:
does it work also with non pbl methods but others?

Reply:
Correct, in order for this to be used for other methods a new provider class implementing the `OrdersProviderInterface` interface must be passed as argument to the `AdyenCancelOrders` virtualType